Packaging update

LiamPattinson opened the pull request:
I've updated the project to use a modern build system and done some general reorganising. It should be ready to publish, though we may need to set a PyPI token first.

- Switched to pyproject.toml build system
- Updated files to match equivalents in latest freegs
- _version now autogenerated
- Moved tests to top-level directory
- Added black auto-commit workflow
- Added auto-publish workflow
- Updated workflow test versions
- General reformatting
- Updated README

LiamPattinson committed:
Packaging Update

- Switched to pyproject.toml build system
- Updated files to match equivalents in latest freegs
- _version now autogenerated
- Moved tests to top-level directory
- Added black auto-commit workflow
- Added auto-publish workflow
- Updated workflow test versions
- General reformatting

LiamPattinson committed:
README update

LiamPattinson committed:
Rename test workflow

ZedThree commented on .github/workflows/publish.yml:
```diff
@@ -0,0 +1,26 @@
+  deploy:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - name: Set up Python
+        uses: actions/setup-python@v2
```
I think there are more recent versions of these actions btw, like v3 and v4 respectively -- worth double-checking and bumping if necessary

LiamPattinson replied:
Good call!

After looking through again, I spotted this in the black workflow:

```
    - uses: actions/checkout@v2
      with:
        ref: ${{ github.head_ref }}
```

It was copied over from a different project. Do you know if the `with` section is necessary?

LiamPattinson replied:
Never mind, found a better way to handle this

ZedThree commented on .github/workflows/tests.yml:
```diff
@@ -10,12 +10,12 @@ on:
     strategy:
       matrix:
-        python-version: [3.5, 3.6, 3.7, 3.8]
+        python-version: ["3.8", "3.9", "3.10"]
```
Might as well add `"3.11"` too!

LiamPattinson committed:
Updated Workflows to use latest versions

LiamPattinson committed:
Using official black workflow

LiamPattinson commented:
I found an official Black action. It doesn't auto-commit like the previous one, but it's much simpler and easy to maintain. Personally, I'd prefer to have the actions complain at me than to automatically commit, but I can revert to the previous workflow if you think that'd be better.

Edit: reverting back to the old version

LiamPattinson committed:
Revert "Using official black workflow"

This reverts commit 5ffd70957b4969ca1bb7eed915cbe8254b880115.

LiamPattinson committed:
Added try/except to version detection in case package is not installed

This can happen if the user tries to use our package without
installing with pip, i.e. by setting PYTHONPATH.
